examples/text-generation/text_generation.ipynb

```diff
@@ -0,0 +1,1263 @@
+{
```

Comment:
Should I restrict the scope of this example notebook to just the metrics included in this PR for now?

Reply:
I would, but I think it depends how soon after this PR you plan to implement the rest of the metrics. if it's your next priority (ETA 1-2 weeks after you merge this PR), then it's probably fine to leave them